packages/cli/src/operations.ts

```diff
@@ -3,52 +3,24 @@ import {
   transformOas3Operations,
   transformPostmanCollectionOperations,
 } from '@stoplight/http-spec';
-import { resolveFile, resolveHttp } from '@stoplight/json-ref-readers';
-import { Resolver } from '@stoplight/json-ref-resolver';
+import { dereference } from 'json-schema-ref-parser';
 import { IHttpOperation } from '@stoplight/types';
-import { parse } from '@stoplight/yaml';
-import fetch from 'node-fetch';
-import * as fs from 'fs';
-import { get, uniq } from 'lodash';
-import { EOL } from 'os';
-import { resolve } from 'path';
+import { get } from 'lodash';
 import type { Spec } from 'swagger-schema-official';
 import type { OpenAPIObject } from 'openapi3-ts';
 import type { CollectionDefinition } from 'postman-collection';
 
-const httpAndFileResolver = new Resolver({
-  resolvers: {
-    https: { resolve: resolveHttp },
-    http: { resolve: resolveHttp },
-    file: { resolve: resolveFile },
-  },
-  parseResolveResult: opts => Promise.resolve({ ...opts, result: parse(opts.result) }),
-});
-
-export async function getHttpOperationsFromResource(file: string): Promise<IHttpOperation[]> {
-  const isRemote = /^https?:\/\//i.test(file);
-  const fileContent = await (isRemote
-    ? fetch(file).then(d => d.text())
-    : fs.promises.readFile(file, { encoding: 'utf8' }));
-
-  return getHttpOperationsFromSpec(fileContent, isRemote ? file : resolve(file));
+export async function getHttpOperationsFromResource(specFilePathOrObject: string | object): Promise<IHttpOperation[]> {
+  return getHttpOperationsFromSpec(specFilePathOrObject);
 }
 
-export async function getHttpOperationsFromSpec(specContent: string, baseUri?: string): Promise<IHttpOperation[]> {
-  const parsedContent = parse(specContent);
-  const { result: resolvedContent, errors } = await httpAndFileResolver.resolve(parsedContent, { baseUri });
-
-  if (errors.length) {
-    const uniqueErrors = uniq(errors.map(error => error.message)).join(EOL);
-    throw new Error(
-      `There's been an error while trying to resolve external references in your document: ${uniqueErrors}`
-    );
-  }
+export async function getHttpOperationsFromSpec(specFilePathOrObject: string | object): Promise<IHttpOperation[]> {
+  const result = await dereference(specFilePathOrObject);
```

Comment:
the `json-schema-ref-parser` package [dereference method](https://apitools.dev/json-schema-ref-parser/docs/ref-parser.html#dereferenceschema-options-callback) expects either a schema object or a file path string. We were passing in the stringified schema object or file contents before. Updated to take either a string --> filePath, or object --> schema object and pass that directly to `dereference` method.